odoo/addons/base/models/ir_asset.py

```diff
@@ -0,0 +1,334 @@
+        for asset in assets.filtered(lambda a: a.sequence < DEFAULT_SEQUENCE):
+            process_path(asset.directive, asset.target, asset.glob)
+
+        # 3. Goes through all addons' manifests.
+        for addon in addons:
+            manifest = manifest_cache.get(addon)
+            if not manifest:
+                continue
+            manifest_assets = manifest.get('assets', {})
+            for path_def in manifest_assets.get(bundle, []):
+                # Default directive: append
+                directive = APPEND_DIRECTIVE
+                target = None
+                if type(path_def) == tuple:
+                    # Additional directive given
+                    if path_def[0] in DIRECTIVES_WITH_TARGET:
+                        directive, target, path_def = path_def
+                    else:
+                        directive, path_def = path_def
+                process_path(directive, target, path_def)
+
+        # 4. Goes through the rest of 'ir.asset' records
+        for asset in assets.filtered(lambda a: a.sequence >= DEFAULT_SEQUENCE):
+            process_path(asset.directive, asset.target, asset.glob)
+
+        return asset_paths
+
+    def get_related_bundle(self, target_path, root_bundle):
+        """
+        Returns the first bundle directly defining a glob matching the target
+        path. This is useful when generating an 'ir.asset' record to override
+        a specific asset and target the right bundle, i.e. the first one
+        defining the target path.
+
+        :param target_path: string: path to match.
+        :root_bundle: string: bundle from which to initiate the search.
+        :returns: the first matching bundle or None
+        """
+        ext = target_path.split('.')[-1]
+
+        js = False
+        css = False
+        xml = False
+
+        if ext in SCRIPT_EXTENSIONS:
+            js = True
+        elif ext in STYLE_EXTENSIONS:
+            css = True
+        elif ext in TEMPLATE_EXTENSIONS:
+            xml = True
+
+        asset_paths = self.get_asset_paths(bundle=root_bundle, css=css, js=js, xml=xml)
+
+        for path, _, bundle in asset_paths:
+            if path == target_path:
+                return bundle
+
+    def _get_asset_domain(self, bundle):
+        """Meant to be overridden to add additional parts to the search domain"""
+        return [('bundle', '=', bundle), ('active', '=', True)]
+
+    def _get_addons_list(self):
+        """
+        Returns the list of addons to take into account when loading assets.
+        Can be overridden to filter the returned list of modules.
+        :returns: string[]: list of module names
+        """
+        if not http.request:
+            return self.env['ir.module.module'].sudo()._installed_sorted()
+        else:
+            return http.module_boot()
+
+    @staticmethod
+    def _get_manifest_cache():
+        return http.addons_manifest
```

Comment:
Static methods are usually avoided in Odoo models.
I don't see any override, do you have a use-case for overriding this?